From 4d417901059e4ff7242756049947d502f77648e9 Mon Sep 17 00:00:00 2001
From: Anson Huang <b20788@freescale.com>
Date: Thu, 3 Sep 2015 00:02:22 +0800
Subject: [PATCH] MLK-11491 ARM: imx: dis-connect vddhigh_in and vddsnvs_in
 for imx6ul

To achieve lower power number in low power idle, on the boards
which has higher vddhigh_in than vddsnvs_in, need to dis-connect
vddhigh_in and vddsnvs_in internally to make vddhigh_in's power
low enough to meet the design target.

Signed-off-by: Anson Huang <b20788@freescale.com>
---
 arch/arm/mach-imx/imx6ul_low_power_idle.S | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm/mach-imx/imx6ul_low_power_idle.S b/arch/arm/mach-imx/imx6ul_low_power_idle.S
index 5db724e..26bb83d 100644
--- a/arch/arm/mach-imx/imx6ul_low_power_idle.S
+++ b/arch/arm/mach-imx/imx6ul_low_power_idle.S
@@ -310,6 +310,11 @@
 	orr	r7, r7, #0x6000
 	str	r7, [r10, #0x150]
 
+	/* disconnect vdd_high_in and vdd_snvs_in */
+	ldr	r7, [r10, #0x150]
+	orr	r7, r7, #0x1000
+	str	r7, [r10, #0x150]
+
 	.endm
 
 	.macro anatop_exit_idle
-- 
1.8.0

